src/bin/update-downloads.rs

```diff
@@ -38,13 +38,23 @@ fn update(conn: &PgConnection) -> QueryResult<()> {
         let rows = version_downloads
             .filter(processed.eq(false))
             .filter(id.gt(m))
+            .filter(downloads.ne(counted))
             .order(id)
             .limit(LIMIT)
             .load(conn)?;
         collect(conn, &rows)?;
         max = rows.last().map(|d| d.id);
     }
 
+    // Anything older than 24 hours ago will be frozen and will not be queried
+    // against again.
+    diesel::update(version_downloads)
+        .set(processed.eq(true))
+        .filter(date.lt(diesel::dsl::date(now)))
```

Comment:
The comment says older than 24 hours ago, but I read this more as "yesterday or older in UTC".  The behavior itself seems fine to me, but the comment confused me a bit at first.  (As you mentioned the old behavior was 2 days ago, so the comment was confusing then as well.)

Other than that, I'm r+.  This logic seems clearer and more efficient now and I think I finally understand how this portion of the logic works!

Reply:
Yeah, 24 hours is kinda wrong, but this matches what records will actually ever be touched again. After midnight UTC we never update old records.